app/context/SearchProvider.tsx

```diff
@@ -0,0 +1,38 @@
+"use client";
+
+import { createContext, useContext, useState } from "react";
+
+type SearchContextType = {
+  search: string;
+  setSearch: (search: string) => void;
+};
+
+export const SearchContext = createContext<SearchContextType>({
```

Comment:
Using a `context` api to handle search data is a good idea, but considering our plans to scale the app and handle a significant amount of data efficiently. react-query for example can handle data for us, including search data, which might let us skip using our context setup.

It could make our code cleaner and easier to manage.